gix-fsck/src/lib.rs

```diff
@@ -80,9 +80,7 @@ where
             match entry_ref.mode.kind() {
                 EntryKind::Tree => {
                     let tree_id = entry_ref.oid.to_owned();
-                    if self.seen.insert(tree_id) {
```

Comment:
I do admit that I don't get how it solves the recursion issue, as I would have thought the `seen` check is to assure it doesn't look at the same tree twice.